rmf_site_editor/src/lib.rs

```diff
@@ -15,6 +14,10 @@ pub use autoload::*;
 pub mod asset_loaders;
 use asset_loaders::*;
 
+// Bevy plugins that are public dependencies, mixing versions won't work for downstream users
+pub use bevy_egui;
+pub use bevy_mod_raycast;
```

Comment:
This is another interesting one that I'm considering pushing further.
If users need to use any of these plugins in their downstream applications they need to also include them in the exact same version in their `Cargo.toml` or might end up with very cryptic errors if two different versions are included in their application vs the site editor (since the two different versions will compile to types with the same name that are actually different and not interchangeable).

So my approach was to just make these dependencies public so users will just import them through the site editor. These two are the bare minimum needed for the workcell editor and probably most applications (`bevy_egui` is needed to create custom widgets, `bevy_mod_raycast` for raycasting for intersections with meshes / ground planes) but I'm happy to make more dependencies public if this approach sounds good.

Reply:
This is a good idea for anything that we're exposing in the public API of the site editor :+1: 

I would just avoid re-exporting any dependencies that aren't explicitly being used in our public API, but I'm sure you know that already.

Reply:
I tried to push this a bit further, specifically I wanted to re-export `bevy` but I'm a bit worried it might not be a good idea.
The main rationale behind it is that `rmf_site_format` depends on `bevy`, if anyone was to write an application that depends on `rmf_site_format` but doesn't use the same version of `bevy` they will have a lot of issues (it actually happened to me in the past when updating to a newer version!).

The approach would have been something like adding in rmf_site_format's [lib.rs](https://github.com/open-rmf/rmf_site/blob/main/rmf_site_format/src/lib.rs) the following:

```
#[cfg(feature = "bevy")]
pub use bevy;
```

Then changing the site editor to do a:

```
// lib.rs
pub use rmf_site_format::bevy;

// Whenever bevy is used
use crate::bevy::prelude::*;
```

That would ensure that both the site editor and all downstream applications use the same version of bevy that `rmf_site_format` was compiled with. However I do not know how this would work with custom features. For example, `rmf_site_format` only depends on bare bevy: https://github.com/open-rmf/rmf_site/blob/983cd3c943e47924c5cf436e30becb3d4cb357c3/rmf_site_format/Cargo.toml#L20

While the site editor needs additional features: https://github.com/open-rmf/rmf_site/blob/983cd3c943e47924c5cf436e30becb3d4cb357c3/rmf_site_editor/Cargo.toml#L34

It is not clear to me if encouraging a pattern of using another crate's public dependency while downstream applications might need to redefine the dependency to add optional features introduces a lot of value, so I'll currently leave as this proposal where the public dependencies are just simple plugins that don't have any optional feature and can just reduce churn when migrating between versions of Bevy / creating a new application that depends on the site editor.